src/transformers/configuration_utils.py

```diff
@@ -667,7 +667,43 @@ def update(self, config_dict: Dict[str, Any]):
         Updates attributes of this class with attributes from ``config_dict``.
 
         Args:
-            config_dict (:obj:`Dict[str, Any]`): Dictionary of attributes that shall be updated for this class.
+            config_dict (:obj:`Dict[str, Any]`): Dictionary of attributes that should be updated for this class.
         """
         for key, value in config_dict.items():
             setattr(self, key, value)
+
+    def update_from_string(self, update_str: str):
+        """
+        Updates attributes of this class with attributes from ``update_str``.
+
+        The expected format is in the form of this example ``key1=10,key2=0.2,key3=true,key4=string`` - i.e. just write
+        ints and floats and strings as is, and for booleans use ``true`` or ``false``.
+
+        The key has to already exist in the config object.
+
+        Args:
+            update_str (:obj:`str`): String with attributes that should be updated for this class.
+
+        """
+
+        d = dict(x.split("=") for x in update_str.split(","))
+        for k, v in d.items():
+            if not hasattr(self, k):
+                raise ValueError(f"key {k} isn't in the original config dict")
+
+            old_v = getattr(self, k)
+            if isinstance(old_v, bool):
+                if v.lower() == "true":
```

Comment:
Since we're going at lengths to parse this, maybe add all truthy values like y, yes and 1?